File: app/src/main/java/com/reco/support/data/CallRecordingRepositoryImpl.kt
package com.reco.support.data

import com.reco.support.data.local.CallRecordingDao
import com.reco.support.domain.model.CallRecording
import com.reco.support.domain.model.SupportStatus
import com.reco.support.domain.repository.CallRecordingRepository
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.map

class CallRecordingRepositoryImpl(
    private val dao: CallRecordingDao
) : CallRecordingRepository {
    override fun observeAll(): Flow<List<CallRecording>> = dao.observeAll().map { list ->
        list.map { it.toDomain() }
    }

    override suspend fun getById(id: Long): CallRecording? = dao.getById(id)?.toDomain()

    override suspend fun insert(recording: CallRecording): Long = dao.insert(recording.toEntity())

    override suspend fun updateNote(id: Long, note: String) = dao.updateNote(id, note)

    override suspend fun updateStatus(id: Long, status: SupportStatus) = dao.updateStatus(id, status.name)
}

File: lib/data/repository/call_recording_repository_impl.dart
import 'package:support_call_recorder/data/local/call_recording_dao.dart';
import 'package:support_call_recorder/domain/model/call_recording.dart';
import 'package:support_call_recorder/domain/model/support_status.dart';
import 'package:support_call_recorder/domain/repository/call_recording_repository.dart';

class CallRecordingRepositoryImpl implements CallRecordingRepository {
  CallRecordingRepositoryImpl(this._dao);
  final CallRecordingDao _dao;

  @override
  Future<List<CallRecording>> getAll() => _dao.list();

  @override
  Future<CallRecording?> getById(int id) => _dao.byId(id);

  @override
  Future<int> insert(CallRecording recording) => _dao.insert(recording);

  @override
  Future<void> updateNote(int id, String note) => _dao.updateNote(id, note);

  @override
  Future<void> updateStatus(int id, SupportStatus status) =>
      _dao.updateStatus(id, status);
}
